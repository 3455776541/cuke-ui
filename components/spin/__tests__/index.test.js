import React from "react";
import assert from "power-assert";
import { render, mount } from "enzyme";
import toJson from "enzyme-to-json";
import Spin from "../index";

describe("<Spin/>", () => {
	it("should render a <Spin/> components", () => {
		const wrapper = render(
			<div>
				<Spin />
				<Spin tip="spin" />
				<Spin indicator="spin" />
			</div>
		);
		expect(toJson(wrapper)).toMatchSnapshot();
	});

	it("should render more size width <Spin/> ", () => {
		const wrapper = render(
			<div>
				<Spin />
				<Spin size="small" />
				<Spin size="large" />
			</div>
		);
		expect(toJson(wrapper)).toMatchSnapshot();
	});

	it("should find cuke-spin classnames", () => {
		const wrapper = mount(
			<div>
				<Spin />
				<Spin size="small" />
				<Spin size="large" />
				<Spin tip="加载中">
					<div>哈哈</div>
				</Spin>
			</div>
		);
		assert(wrapper.find(".cuke-spin").length >= 1);
		assert(wrapper.find(".cuke-spin-container").length >= 1);
		assert(wrapper.find(".cuke-spin-tip").length >= 1);
		assert(wrapper.find(".cuke-spin-blur").length >= 1);
		assert(wrapper.find(".cuke-spin-small").length >= 1);
		assert(wrapper.find(".cuke-spin-large").length >= 1);
	});
});
