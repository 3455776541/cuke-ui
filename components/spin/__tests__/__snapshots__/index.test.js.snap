// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Spin/> should render a <Spin/> components 1`] = `
<div>
  <div
    class="cuke-spin cuke-spin-spinning"
  >
    <span
      class="cuke-spin-indicator"
    >
      <svg
        attr="[object Object]"
        fill="none"
        height="1em"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="1em"
      >
        <line
          x1="12"
          x2="12"
          y1="2"
          y2="6"
        />
        <line
          x1="12"
          x2="12"
          y1="18"
          y2="22"
        />
        <line
          x1="4.93"
          x2="7.76"
          y1="4.93"
          y2="7.76"
        />
        <line
          x1="16.24"
          x2="19.07"
          y1="16.24"
          y2="19.07"
        />
        <line
          x1="2"
          x2="6"
          y1="12"
          y2="12"
        />
        <line
          x1="18"
          x2="22"
          y1="12"
          y2="12"
        />
        <line
          x1="4.93"
          x2="7.76"
          y1="19.07"
          y2="16.24"
        />
        <line
          x1="16.24"
          x2="19.07"
          y1="7.76"
          y2="4.93"
        />
      </svg>
    </span>
  </div>
  <div
    class="cuke-spin cuke-spin-spinning"
  >
    <span
      class="cuke-spin-indicator"
    >
      <svg
        attr="[object Object]"
        fill="none"
        height="1em"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="1em"
      >
        <line
          x1="12"
          x2="12"
          y1="2"
          y2="6"
        />
        <line
          x1="12"
          x2="12"
          y1="18"
          y2="22"
        />
        <line
          x1="4.93"
          x2="7.76"
          y1="4.93"
          y2="7.76"
        />
        <line
          x1="16.24"
          x2="19.07"
          y1="16.24"
          y2="19.07"
        />
        <line
          x1="2"
          x2="6"
          y1="12"
          y2="12"
        />
        <line
          x1="18"
          x2="22"
          y1="12"
          y2="12"
        />
        <line
          x1="4.93"
          x2="7.76"
          y1="19.07"
          y2="16.24"
        />
        <line
          x1="16.24"
          x2="19.07"
          y1="7.76"
          y2="4.93"
        />
      </svg>
    </span>
  </div>
  <div
    class="cuke-spin cuke-spin-spinning"
  >
    <span
      class="cuke-spin-indicator"
    >
      spin
    </span>
  </div>
</div>
`;

exports[`<Spin/> should render more size width <Spin/>  1`] = `
<div>
  <div
    class="cuke-spin cuke-spin-spinning"
  >
    <span
      class="cuke-spin-indicator"
    >
      <svg
        attr="[object Object]"
        fill="none"
        height="1em"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="1em"
      >
        <line
          x1="12"
          x2="12"
          y1="2"
          y2="6"
        />
        <line
          x1="12"
          x2="12"
          y1="18"
          y2="22"
        />
        <line
          x1="4.93"
          x2="7.76"
          y1="4.93"
          y2="7.76"
        />
        <line
          x1="16.24"
          x2="19.07"
          y1="16.24"
          y2="19.07"
        />
        <line
          x1="2"
          x2="6"
          y1="12"
          y2="12"
        />
        <line
          x1="18"
          x2="22"
          y1="12"
          y2="12"
        />
        <line
          x1="4.93"
          x2="7.76"
          y1="19.07"
          y2="16.24"
        />
        <line
          x1="16.24"
          x2="19.07"
          y1="7.76"
          y2="4.93"
        />
      </svg>
    </span>
  </div>
  <div
    class="cuke-spin cuke-spin-spinning"
  >
    <span
      class="cuke-spin-indicator cuke-spin-small"
    >
      <svg
        attr="[object Object]"
        fill="none"
        height="1em"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="1em"
      >
        <line
          x1="12"
          x2="12"
          y1="2"
          y2="6"
        />
        <line
          x1="12"
          x2="12"
          y1="18"
          y2="22"
        />
        <line
          x1="4.93"
          x2="7.76"
          y1="4.93"
          y2="7.76"
        />
        <line
          x1="16.24"
          x2="19.07"
          y1="16.24"
          y2="19.07"
        />
        <line
          x1="2"
          x2="6"
          y1="12"
          y2="12"
        />
        <line
          x1="18"
          x2="22"
          y1="12"
          y2="12"
        />
        <line
          x1="4.93"
          x2="7.76"
          y1="19.07"
          y2="16.24"
        />
        <line
          x1="16.24"
          x2="19.07"
          y1="7.76"
          y2="4.93"
        />
      </svg>
    </span>
  </div>
  <div
    class="cuke-spin cuke-spin-spinning"
  >
    <span
      class="cuke-spin-indicator cuke-spin-large"
    >
      <svg
        attr="[object Object]"
        fill="none"
        height="1em"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="1em"
      >
        <line
          x1="12"
          x2="12"
          y1="2"
          y2="6"
        />
        <line
          x1="12"
          x2="12"
          y1="18"
          y2="22"
        />
        <line
          x1="4.93"
          x2="7.76"
          y1="4.93"
          y2="7.76"
        />
        <line
          x1="16.24"
          x2="19.07"
          y1="16.24"
          y2="19.07"
        />
        <line
          x1="2"
          x2="6"
          y1="12"
          y2="12"
        />
        <line
          x1="18"
          x2="22"
          y1="12"
          y2="12"
        />
        <line
          x1="4.93"
          x2="7.76"
          y1="19.07"
          y2="16.24"
        />
        <line
          x1="16.24"
          x2="19.07"
          y1="7.76"
          y2="4.93"
        />
      </svg>
    </span>
  </div>
</div>
`;
